### PR TITLE
doc: Add README for resolver-tests

### PR DESCRIPTION
**Pull request by linyihai**
### What does this PR try to resolve?

This collect the comments from @Eh2406 about the resolver-tests and add a README to it.

### Additional information

Maybe Fixed https://github.com/rust-lang/cargo/issues/13319
See  https://github.com/rust-lang/cargo/pull/11938#issuecomment-1868426431

### CODE REVIEW

File: crates/resolver-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# resolver-tests
+
+## The aim
+
+This crate aims to test the resolution of cargo's resolver. It implements a [SAT solver](https://en.wikipedia.org/wiki/SAT_solver) to compare resolution of cargo's resolver.
```
Comment (Eh2406) [inline crates/resolver-tests/README.md:5]:
```
- to compare resolution of cargo's resolver.
+ to compare with the resolution of cargo's resolver.
```

File: crates/resolver-tests/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# resolver-tests
+
+## The aim
+
+This crate aims to test the resolution of cargo's resolver. It implements a [SAT solver](https://en.wikipedia.org/wiki/SAT_solver) to compare resolution of cargo's resolver.    
+This ensures that cargo's dependency resolution is valid on the [SAT](https://en.wikipedia.org/wiki/Boolean_satisfiability_problem).
```
Comment (Eh2406) [inline crates/resolver-tests/README.md:6]:
```
- on the [SAT]
+ by lowering to [SAT problem]
```
not sure exactly what the clearest wording is here.

File: crates/resolver-tests/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# resolver-tests
+
+## The aim
+
+This crate aims to test the resolution of cargo's resolver. It implements a [SAT solver](https://en.wikipedia.org/wiki/SAT_solver) to compare resolution of cargo's resolver.    
+This ensures that cargo's dependency resolution is valid on the [SAT](https://en.wikipedia.org/wiki/Boolean_satisfiability_problem).   
+
+## About the test
+
+The cargo's resolver is very sensitive to what order it tries to evaluate constraints.  This makes it incredibly difficult
```
Comment (Eh2406) [inline crates/resolver-tests/README.md:10]: > The cargo's resolver

Nit: "The resolver in cargo" or "Cargo's resolver"

File: crates/resolver-tests/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+## The aim
+
+This crate aims to test the resolution of cargo's resolver. It implements a [SAT solver](https://en.wikipedia.org/wiki/SAT_solver) to compare resolution of cargo's resolver.    
+This ensures that cargo's dependency resolution is valid on the [SAT](https://en.wikipedia.org/wiki/Boolean_satisfiability_problem).   
+
+## About the test
+
+The cargo's resolver is very sensitive to what order it tries to evaluate constraints.  This makes it incredibly difficult     
+to be sure that a handful of tests actually covers all the important permutations of decision-making. The tests not only needs    
+to hit all the corner cases, it needs to try all of the orders of evaluation. So we use fuzz testing to cover more permutations as possible.
```
Comment (Eh2406) [inline crates/resolver-tests/README.md:12]: > to cover more permutations as possible.

"to cover more permutations" or "to cover more of the possible permutations"

**Review by Eh2406 — CHANGES_REQUESTED** [review]
Thank you so much for this documentation work. I have A couple of Nits about grammar. But overall it looks like a good next step.

File: crates/resolver-tests/README.md (hunk 3 above)
Comment (linyihai) [reply crates/resolver-tests/README.md]: `Cargo's resolver` seems more succinct

**Comment by Eh2406** [comment]
Thank you!

@bors r+

**Comment by weihanglo** [comment]
@Eh2406 I guess you need to approve the PR. bors is complaining
